src/mumble/GlobalShortcut_xdp.cpp

```diff
@@ -0,0 +1,200 @@
+        m_ids[i] = shortcut->objectName();
+        Global::get().s.qlShortcuts << ourShortcut;
+        ++i;
+    }
+
+    QDBusArgument arg;
+    arg << initialShortcuts;
+    QDBusPendingReply<QDBusObjectPath> reply = s_shortcutsInterface->CreateSession({
+        { QLatin1String("session_handle_token"), "Mumble" },
+        { QLatin1String("handle_token"), QLatin1String("mumble") },
+        { QLatin1String("shortcuts"), QVariant::fromValue(arg) },
+    });
+    reply.waitForFinished();
+    if (reply.isError()) {
+        qWarning() << "Couldn't get reply";
+        qWarning() << "Error:" << reply.error().message();
+    } else {
+        QDBusConnection::sessionBus().connect(QString(),
+                                            reply.value().path(),
+                                            QLatin1String("org.freedesktop.portal.Request"),
+                                            QLatin1String("Response"),
+                                            this,
+                                            SLOT(gotGlobalShortcutsCreateSessionResponse(uint,QVariantMap)));
+    }
+}
+
+void GlobalShortcutXdp::shortcutActivated(const QDBusObjectPath &session_handle, const QString &shortcut_id, qulonglong timestamp, const QVariantMap &options)
+{
+    Q_UNUSED(session_handle);
+    Q_UNUSED(timestamp);
+    Q_UNUSED(options);
+
+    for (GlobalShortcut *shortcut : qmShortcuts) {
+        if (shortcut_id == shortcut->objectName()) {
+            shortcut->triggered(true, shortcut_id);
+            shortcut->down(shortcut_id);
+        }
+    }
+}
+
+void GlobalShortcutXdp::shortcutDeactivated(const QDBusObjectPath &session_handle, const QString &shortcut_id, qulonglong timestamp, const QVariantMap &options)
+{
+    Q_UNUSED(session_handle);
+    Q_UNUSED(timestamp);
+    Q_UNUSED(options);
+
+    for (GlobalShortcut *shortcut : qmShortcuts) {
+        if (shortcut_id == shortcut->objectName()) {
+            shortcut->triggered(false, shortcut_id);
+        }
+    }
+}
+
+void GlobalShortcutXdp::shortcutsConfigured(const QDBusObjectPath &session_handle, const XdpShortcuts &shortcuts)
+{
+    Q_UNUSED(session_handle);
+    if (m_shortcuts != shortcuts) {
+        m_shortcuts = shortcuts;
+        emit shortcutsChanged();
+    }
+}
+
+void GlobalShortcutXdp::run() {
+	// 🤘🤪
+}
+
+bool GlobalShortcutXdp::canDisable() {
+	return false;
+}
+
+GlobalShortcutXdp::ButtonInfo GlobalShortcutXdp::buttonInfo(const QVariant &v) {
+	ButtonInfo info;
+	bool ok;
+	unsigned int key = v.toUInt(&ok);
+	if (!ok) {
+		return info;
+	}
+
+    info.device       = tr("Desktop");
+    info.devicePrefix = QString();
+    const QString id = m_ids[key];
+    for (const XdpShortcut &x : m_shortcuts) {
+        if (x.first == id) {
+            info.name = x.second["trigger_description"].toString();
+        }
+    }
+	return info;
+}
+
+static QString parentWindowId()
+{
+    if (EnvUtils::waylandIsUsed()) {
+        // TODO
+        return {};
```

Comment:
Just marking this so we don't lose track of this TODO

Reply:
So to address this TODO we'll need to use an interface like this:
https://invent.kde.org/libraries/xdg-portal-test-kde/-/merge_requests/18

This means depending on Qt::GuiPrivate and libwaylandclient. If you are interested in this I can include it in this PR.

Reply:
Hm. This seems rather hacky. What are the implications of simply leaving the implementation as it is? Aka: what exactly is the parent window ID being used for?
Will this perhaps only become relevant when dealing with multiple Mumble instances?

Reply:
The parent window is simply to inform the portal who is calling this and which window the dialog should be on top.

The dialog will appear on top and focussed anyway, so it shouldn't be a big deal. FWIW, there will be public API for this in Qt 6 (6.5, if I'm not mistaken). If you don't want to depend on private API, just delaying until Qt 6 is a good option.

Reply:
Okay in that case it indeed seems a good idea to simply delay this :+1:

Reply:
#6516